File: VB/WindowsApplication53/MyPivotGrid.vb
Imports DevExpress.XtraPivotGrid
Imports DevExpress.XtraPivotGrid.Data
Imports DevExpress.XtraPivotGrid.ViewInfo
Imports DevExpress.XtraEditors.Filtering
Imports DevExpress.XtraEditors.Repository

Namespace WindowsApplication53

    Public Class MyPivotGridControl
        Inherits PivotGridControl

        Public Sub New()
            MyBase.New()
        End Sub

        Public Sub New(ByVal viewInfoData As PivotGridViewInfoData)
            MyBase.New(viewInfoData)
        End Sub

        Protected Overrides Function CreateData() As PivotGridViewInfoData
            Return New MyPivotGridViewInfoData(Me)
        End Function
    End Class

    Public Class MyPivotGridViewInfoData
        Inherits PivotGridViewInfoData
        Implements IFilteredComponent

        Public Sub New()
            MyBase.New()
        End Sub

        Public Sub New(ByVal control As IViewInfoControl)
            MyBase.New(control)
        End Sub

#Region "IFilteredComponent Members"
        Private Overloads Function CreateFilterColumnCollection() As IBoundPropertyCollection Implements IFilteredComponent.CreateFilterColumnCollection
            Dim collection As IBoundPropertyCollection = New FieldFilterColumnCollection(Fields, Prefilter.PrefilterColumnNames, MenuManager)
            For Each column As FieldFilterColumn In collection
                UpdateFieldFilterColumn(column)
            Next

            Return collection
        End Function

        Private Sub UpdateFieldFilterColumn(ByVal filterColumn As FieldFilterColumn)
            Dim comboBox As RepositoryItemComboBox = TryCast(filterColumn.ColumnEditor, RepositoryItemComboBox)
            If comboBox IsNot Nothing Then
                comboBox.AutoComplete = False
            End If
        End Sub
#End Region
    End Class
End Namespace

File: CS/WindowsApplication53/MyPivotGrid.cs
﻿using System;
using System.Collections.Generic;
using System.Text;
using DevExpress.XtraPivotGrid;
using DevExpress.XtraPivotGrid.Data;
using DevExpress.XtraPivotGrid.ViewInfo;
using DevExpress.XtraEditors.Filtering;
using DevExpress.XtraEditors.Repository;

namespace WindowsApplication53
{
    public class MyPivotGridControl : PivotGridControl
    {
        public MyPivotGridControl() : base() { }
        public MyPivotGridControl(PivotGridViewInfoData viewInfoData) : base(viewInfoData) { }

        protected override PivotGridViewInfoData CreateData()
        {
            return new MyPivotGridViewInfoData(this);
        }
    }

    public class MyPivotGridViewInfoData : PivotGridViewInfoData, IFilteredComponent 
    {
        public MyPivotGridViewInfoData() : base() { }
        public MyPivotGridViewInfoData(IViewInfoControl control) : base(control) { }


        #region IFilteredComponent Members

        FilterColumnCollection IFilteredComponent.CreateFilterColumnCollection()
        {
            FilterColumnCollection collection = new FilterColumnCollection();
            for (int i = 0; i < Fields.Count; i++)
            {
                if (Fields[i].CanShowInPrefilter)
                {
                    FieldFilterColumn filterColumn = new FieldFilterColumn(Fields[i], MenuManager);
                    UpdateFieldFilterColumn(filterColumn);
                    collection.Add(filterColumn); 
                }
            }
            return collection;
        }

        private void UpdateFieldFilterColumn(FieldFilterColumn filterColumn)
        {
            RepositoryItemComboBox comboBox = filterColumn.ColumnEditor as RepositoryItemComboBox;
            if (comboBox != null)
            {
                comboBox.AutoComplete = false;
            }
        }
        #endregion

        
    }

}
